src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeTracker.cs

```diff
@@ -111,7 +111,8 @@ public Task StartFileChangeListeningAsync()
             {
                 try
                 {
-                    return await ((IVsAsyncFileChangeEx)_fileChangeService).AdviseFileChangeAsync(_filePath, _fileChangeFlags, this).ConfigureAwait(false);
+                    // TODO: Should we pass in cancellationToken here insead of CancellationToken.None?
+                    return await ((IVsAsyncFileChangeEx)_fileChangeService).AdviseFileChangeAsync(_filePath, _fileChangeFlags, this, CancellationToken.None).ConfigureAwait(false);
```

Comment:
Yeah, this is fine since if you look at the call to _fileChangeCookie.GetValueAsync it's passing in CancellationToken.None anyways. Passing in cancellationToken would have had the same effect since it would have been None in practice. That said any attempt to make this cancellation safe is going to require an audit of the entire type which is definitely out of scope.

Reply:
(I admit it's not entirely clear to me why this is using AsyncLazy in the first place...)